sound/soc/sof/control.c

```diff
@@ -429,12 +429,6 @@ int snd_sof_bytes_ext_get(struct snd_kcontrol *kcontrol,
 		(struct snd_ctl_tlv __user *)binary_data;
 	size_t data_size;
 
-	/*
-	 * Decrement the limit by ext bytes header size to
-	 * ensure the user space buffer is not exceeded.
-	 */
-	size -= sizeof(const struct snd_ctl_tlv);
```

Comment:
formally this is correct, yes. But shouldn't this value actually be used to verify the size of the available user-space buffer?

Reply:
@lyakh I have no idea what this 'size' parameter is supposed to indicate. I don't know why we don't use it? @lgirdwood @juimonen this was modified multiple times and something's not right.

Reply:
hmm also puzzled by this code... if we remove the size, should it be removed also from functions params?
@lgirdwood any recollection of this code? I've not worked exactly with this code part...

Reply:
Hmm, I'm suprised we don't get a warning about unused parameter after this patch. But beyond that, I think we are missing a check on "size". We only check the size of the IPC message struct doesn't exceed the limit set in topology, but we should also check the size versus what user-space has passed to ioctl (in sound/core/control.c:snd_ctl_tlv_ioctl()).

Reply:
Agree, we need to between check things here, the code is suspicious at best.